sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -268,6 +268,15 @@ private[spark] object QueryCompilationErrors {
       s"(valid range is [1, $size])", t.origin.line, t.origin.startPosition)
   }
 
+  def groupByPositionRefersToAggregateFunctionError(
+      index: Int,
+      expr: Expression,
+      t: TreeNode[_]): Throwable = {
+    new AnalysisException(s"GROUP BY $index refers to an expression that is or contains " +
+      "an aggregate function. Aggregate functions are not allowed in GROUP BY, " +
+      s"but got ${expr.sql}", t.origin.line, t.origin.startPosition)
```

Comment:
Instead of using `t.origin.line` and `t.origin.startPosition`, we can wrap the code with `withPosition` so we can remove `t` here (see comment in Anayzer).

Reply:
> Instead of using `t.origin.line` and `t.origin.startPosition`, we can wrap the code with `withPosition` so we can remove `t` here (see comment in Anayzer).

Nice suggestion, thanks